CPP_STYLE_GUIDE.md

```diff
@@ -200,6 +201,11 @@ void FooConsumer(std::unique_ptr<Foo> ptr);
 
 Never use `std::auto_ptr`. Instead, use `std::unique_ptr`.
 
+### Avoid non-const references
+
+Using non-const references often obscures which values are changed by an
```

Comment:
Maybe we should give a few more examples about this, instead of going with a firm phrasing about pointers over non-const references all the time? Some of the scenarios include:

- Input parameters
- Output parameters & return values
- Class members
- Places where polymorphism is wanted v.s. not necessary

IMO it's more important to have a consistent style for different scenarios than to be pedantic about which style we prefer in general.

Reply:
@joyeecheung Could you take a look at 8daa18eb3744c93222aae9a9cf9f0af300656144? :)